pkg/apis/serving/k8s_validation_test.go

```diff
@@ -513,6 +507,138 @@ func TestPodSpecMultiContainerValidation(t *testing.T) {
 	}
 }
 
+func TestPodSpecFeatureValidation(t *testing.T) {
+	featureData := []struct {
+		name        string
+		podSpec     corev1.PodSpec
+		featureSpec corev1.PodSpec
+		cfgOpts     []configOption
+		err         *apis.FieldError
+	}{{
+		name: "Affinity",
+		podSpec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Image: "busybox",
+			}},
+		},
+		featureSpec: corev1.PodSpec{
+			Affinity: &corev1.Affinity{
+				NodeAffinity: &corev1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+						NodeSelectorTerms: []corev1.NodeSelectorTerm{{
+							MatchExpressions: []corev1.NodeSelectorRequirement{{
+								Key:      "failure-domain.beta.kubernetes.io/zone",
+								Operator: "In",
+								Values:   []string{"us-east1-b"},
+							}},
+						}},
+					},
+				},
+			},
+		},
+		err: &apis.FieldError{
+			Message: "must not set the field(s)",
+			Paths:   []string{"affinity"},
+		},
+		cfgOpts: []configOption{withFeature("PodSpecAffinity")},
+	}, {
+		name: "NodeSelector",
+		podSpec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Image: "busybox",
+			}},
+		},
+		featureSpec: corev1.PodSpec{
+			NodeSelector: map[string]string{
+				"kubernetes.io/arch": "amd64",
+			},
+		},
+		err: &apis.FieldError{
+			Message: "must not set the field(s)",
+			Paths:   []string{"nodeSelector"},
+		},
+		cfgOpts: []configOption{withFeature("PodSpecNodeSelector")},
+	}, {
+		name: "Tolerations",
+		podSpec: corev1.PodSpec{
+			Containers: []corev1.Container{{
+				Image: "busybox",
+			}},
+		},
```

Comment:
all the `podSpecs` are the same so I wonder if we can drop as an input

Reply:
most likely. I was thinking there may be cases where it would be needed but clearly I didn't have one :)